src/Microsoft.Extensions.Http/DependencyInjection/HttpClientFactoryOptionsSetup.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Http;
+using Microsoft.Extensions.Http.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    internal class HttpClientFactoryOptionsSetup : IConfigureNamedOptions<HttpClientFactoryOptions>
+    {
+        private readonly LoggingHttpMessageHandlerPolicy _loggingPolicy;
+
+        public HttpClientFactoryOptionsSetup(LoggingHttpMessageHandlerPolicy loggingPolicy)
+        {
+            if (loggingPolicy == null)
+            {
+                throw new ArgumentNullException(nameof(loggingPolicy));
+            }
+
+            _loggingPolicy = loggingPolicy;
+        }
+
+        public void Configure(HttpClientFactoryOptions options)
+        {
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            options.HttpMessageHandlerPolicies.Stages.Add(DefaultStages.Default);
+            options.HttpMessageHandlerPolicies.Stages.Add(DefaultStages.Reliability);
+            options.HttpMessageHandlerPolicies.Stages.Add(DefaultStages.ServiceDiscovery);
+            options.HttpMessageHandlerPolicies.Stages.Add(DefaultStages.Security);
+            options.HttpMessageHandlerPolicies.Stages.Add(DefaultStages.Diagnostics);
```

Comment:
This order makes the most sense to me. 

I personally list this approach because I think we need to try and find a consensus ordering of these kinds of policies and recommend that by default. Consider that maybe 4-5 packages deliver policies you want, and some of them are 3rd party. 

That's what I have in mind by doing things like this.

It would be **techically** easy to just assign everything a numeric order. But it may be harder to get the E2E right without a lot of tweaking.

Reply:
I'm struggling to see why we need this IIS style pipeline for http client. What concrete problems are we solving by pre-defining those very opinionated stages into the core abstraction?

Reply:
Do we expect users to have handlers brought in from random places outside their control? That seems to be the major argument in favor of pipeline stages (it's why IIS has them, for example). I'd envisioned this feature much like middleware, where the app remains in control of the order of everything via the set-up methods. We don't do this for inbound middleware so it seems weird that "outbound" middleware would do it.

Reply:
It also assumes we know what the stages should be and this seems completely premature at this point. Stages can also be built on top of a linear pipeline (see katana stage markers)